src/app/content/highlights/hooks/updateHighlight.ts

```diff
@@ -15,20 +13,22 @@ export const hookBody: ActionHookBody<typeof updateHighlight> =
     try {
       await highlightClient.updateHighlight(payload);
     } catch (error) {
-      Sentry.captureException(error);
-
       const oldColor = meta.preUpdateData.highlight.color;
       const oldAnnotation = meta.preUpdateData.highlight.annotation;
 
       if (oldColor === payload.highlight.color && oldAnnotation === payload.highlight.annotation) { return; }
 
+      dispatch(updateHighlight(meta.preUpdateData, {...meta, revertingAfterFailure: true}));
+
+      if (error instanceof CustomApplicationError) {
+        throw error;
+      }
+
       if (payload.highlight.color && oldColor !== payload.highlight.color) {
-        dispatch(addToast(toastMessageKeys.higlights.failure.update.color, {destination}));
+        throw new HighlightUpdateColorError({ destination });
```

Comment:
i'm just now thinking... shouldn't each error type always know what its destination should be? does that have to be defined on the fly?

Reply:
In some cases `destination` is taken from the selector (we are checking if MH popup is open or not) so we could replace this with:
```
if (destination === 'myHighlights) {
  throw ...
} else {
  throw ...
}
```
I think the current solution is easier to ready